tests/suite/reconfig_test.go

```diff
@@ -0,0 +1,614 @@
+				startTime,
+				&endTime,
+				queryRangeStep,
+			),
+		).WithTimeout(metricExistTimeout).WithPolling(metricExistPolling).Should(Succeed())
+
+		getEndTime := func() time.Time { return endTime }
+		noOpModifier := func() {}
+
+		queries := []string{
+			fmt.Sprintf(`container_memory_usage_bytes{pod="%s",container="nginx-gateway"}`, ngfPodName),
+			// We don't need to check all nginx_gateway_fabric_* metrics, as they are collected at the same time
+			fmt.Sprintf(`nginx_gateway_fabric_nginx_reloads_total{pod="%s"}`, ngfPodName),
+		}
+
+		for _, q := range queries {
+			Eventually(
+				framework.CreateMetricExistChecker(
+					promInstance,
+					q,
+					getEndTime,
+					noOpModifier,
+				),
+			).WithTimeout(metricExistTimeout).WithPolling(metricExistPolling).Should(Succeed())
+		}
+
+		checkContainerLogsForErrors(ngfPodName, false)
+
+		reloadCount, err := framework.GetReloadCount(promInstance, ngfPodName)
+		Expect(err).ToNot(HaveOccurred())
+		fmt.Println(reloadCount)
+
+		reloadAvgTime, err := framework.GetReloadAvgTime(promInstance, ngfPodName)
+		Expect(err).ToNot(HaveOccurred())
+		fmt.Println(reloadAvgTime)
+
+		reloadBuckets, err := framework.GetReloadBuckets(promInstance, ngfPodName)
+		Expect(err).ToNot(HaveOccurred())
+		fmt.Println(reloadBuckets)
+
+		eventsCount, err := framework.GetEventsCount(promInstance, ngfPodName)
+		Expect(err).ToNot(HaveOccurred())
+		fmt.Println(eventsCount)
+
+		eventsAvgTime, err := framework.GetEventsAvgTime(promInstance, ngfPodName)
+		Expect(err).ToNot(HaveOccurred())
+		fmt.Println(eventsAvgTime)
+
+		eventsBuckets, err := framework.GetEventsBuckets(promInstance, ngfPodName)
+		Expect(err).ToNot(HaveOccurred())
+		fmt.Println(eventsBuckets)
+
+		logs, err := resourceManager.GetPodLogs(ngfNamespace, ngfPodName, &core.PodLogOptions{
+			Container: "nginx-gateway",
+		})
+		Expect(err).ToNot(HaveOccurred())
+
+		timeToReadyTotal, err := calculateTimeToReadyTotal(logs, timeToReadyStartingLogSubstring)
+		Expect(err).ToNot(HaveOccurred())
+
+		fmt.Println(timeToReadyTotal)
+
+		timeToReadyAvgSingle, err := calculateTimeToReadyAverage(logs)
+		Expect(err).ToNot(HaveOccurred())
+		fmt.Println(timeToReadyAvgSingle)
+
+		results := reconfigTestResults{
+			Name:                 testName,
+			EventsBuckets:        eventsBuckets,
+			ReloadBuckets:        reloadBuckets,
+			NumResources:         resourceCount,
+			TimeToReadyTotal:     timeToReadyTotal,
+			TimeToReadyAvgSingle: timeToReadyAvgSingle,
+			NGINXReloads:         int(reloadCount),
+			NGINXReloadAvgTime:   int(reloadAvgTime),
+			EventsCount:          int(eventsCount),
+			EventsAvgTime:        int(eventsAvgTime),
+		}
+
+		err = writeReconfigResults(outFile, results)
+		Expect(err).ToNot(HaveOccurred())
+
+		cleanupResources(resourceCount)
+	}
+
+	// Test 1 - Resources exist before start-up
+	It("test 1 - 30 resources", func() {
```

Comment:
Time to run a 30 numResources test takes around 10 minutes, I will test how long the 150 numResources test takes but it should be 4-5x that time.

The majority of that time is spent in the `cleanupResources` function. There was a possible suggestion of not tearing down NGF and redeploying it between tests but instead scaling NGF down, but that part of the test does not represent a substantial part of the total test time and would possibly cause issues with prometheus queries, so I avoided that.

Reply:
time to run a 150 numResources test takes around 36 minutes.